libc/utils/MPFRWrapper/MPFRUtils.cpp

```diff
@@ -790,6 +792,37 @@ template void explain_unary_operation_single_output_error<float16>(
     Operation op, float16, float16, double, RoundingMode);
 #endif
 
+template <typename OutType, typename InType>
+void explain_unary_narrower_operation_single_output_error(
```

Comment:
Maybe completely replacing `explain_unary_operation_single_output_error` with this version?